libs/dal/src/repositories/environment/environment.repository.ts

```diff
@@ -51,12 +51,21 @@ export class EnvironmentRepository extends BaseRepository<EnvironmentDBModel, En
     );
   }
 
-  async findByApiKey(key: string) {
+  async findByApiKey(hash: string) {
     return await this.findOne({
-      'apiKeys.key': key,
+      'apiKeys.hash': hash,
     });
   }
 
+  /*
+   * backward compatibility -
+   * * delete findByApiKeyBackwardCompatibility
+   * * use findByApiKey instead once encrypt-api-keys-migration executed
+   */
+  async findByApiKeyBackwardCompatibility({ key, hash }: { key: string; hash: string }) {
+    return await this.findOne({ $or: [{ 'apiKeys.key': key }, { 'apiKeys.hash': hash }] });
+  }
```

Comment:
Some self-hosted Customers may never execute this migration under the current status-quo. So we'll need a way to support both flows until we have a forced deprecation of unencrypted API keys.

I think we should rename this method to `findByApiKey` and remove the other one, and create a Linear ticket to address removal of this sometime in the distant future after we have notified self-hosted Customers to ensure they run this migration.

Reply:
This is another way of doing it, feels more clean to me, ill update the code :) 
We already have a dedicated ticket  💪